### PR TITLE
"Shotgun Shell Pouch/Belt" can be refilled with any magazines (with loose ammo)

### PR DESCRIPTION
**Pull request by BobSlim**
## About The Pull Request

Title. When you click a shotgun shell pouch/belt with a magazine in hand (any magazine, including boxes of ammo), you can fill them with loose rounds.

Ideally, there would be something in /storage to wrap around "loose ammo holder", but I have no idea how to go about doing that. Sounds like one of those fancy refactors.

## Why It's Good For The Game

Someone was complaining about the difficulty of filling up mosin nagant ammo pouches.

## Changelog
:cl:
tweak: You can refill shotgun shell pouches and belts with any magazine/ammo box in hand.
/:cl:

### CODE REVIEW

File: code/game/objects/items/storage/belt.dm [hunk 1]
```diff
@@ -328,8 +328,12 @@
 
 
 /obj/item/storage/belt/shotgun/attackby(obj/item/I, mob/user, params)
-	if(istype(I, /obj/item/ammo_magazine/shotgun))
-		var/obj/item/ammo_magazine/shotgun/M = I
+	
+	if(istype(I, /obj/item/ammo_magazine/handful))
```
Comment (BobSlim) [inline code/game/objects/items/storage/belt.dm:332]: is this the best way to have handfuls have regular functionality, but other mags to empty?

File: code/game/objects/items/storage/pouch.dm [hunk 2]
```diff
@@ -555,30 +555,30 @@
...
+				return
 
-		if(!do_after(user, 15, TRUE, src, BUSY_ICON_GENERIC))
-			return
 
-		for(var/x in 1 to (storage_slots - length(contents)))
-			if(!handle_item_insertion(M.create_handful(), 1, user))
-				break
+			to_chat(user, "<span class='notice'>You start refilling [src] with [M].</span>")
+			if(!do_after(user, 15, TRUE, src, BUSY_ICON_GENERIC))
```
Comment (psykzz) [inline code/game/objects/items/storage/pouch.dm:572]:
```suggestion
			if(!do_after(user, 1.5 SECONDS, TRUE, src, BUSY_ICON_GENERIC))
```

File: code/game/objects/items/storage/pouch.dm [hunk 3]
```diff
@@ -555,30 +555,30 @@
...
 
-		playsound(user.loc, "rustle", 15, 1, 6)
-		to_chat(user, "<span class='notice'>You refill [src] with [M].</span>")
+			for(var/x in 1 to (storage_slots - length(contents)))
+				var/cont = handle_item_insertion(M.create_handful(), 1, user)
+				if(!cont)
+					break
 
-		return TRUE
+			playsound(user.loc, "rustle", 15, 1, 6)
```
Comment (psykzz) [inline code/game/objects/items/storage/pouch.dm:580]:
```suggestion
			playsound(user.loc, "rustle", 15, TRUE, 6)
```

File: code/game/objects/items/storage/belt.dm [hunk 4]
```diff
@@ -328,32 +328,34 @@
...
+			to_chat(user, "<span class='notice'>You start refilling [src] with [M].</span>")
+			if(!do_after(user, 15, TRUE, src, BUSY_ICON_GENERIC))
+				return
+
+			for(var/x in 1 to (storage_slots - length(contents)))
+				var/cont = handle_item_insertion(M.create_handful(), 1, user)
+				if(!cont)
+					break
+
+			playsound(user.loc, "rustle", 15, 1, 6)
```
Comment (psykzz) [inline code/game/objects/items/storage/belt.dm:354]:
```suggestion
			playsound(user.loc, "rustle", 15, TRUE, 6)
```

File: code/game/objects/items/storage/belt.dm [hunk 5]
```diff
@@ -328,32 +328,34 @@
...
+				to_chat(user, "<span class='warning'>[M] is empty.</span>")
+				return
+
+			if(length(contents) >= storage_slots)
+				to_chat(user, "<span class='warning'>[src] is full.</span>")
+				return
+
+
+			to_chat(user, "<span class='notice'>You start refilling [src] with [M].</span>")
+			if(!do_after(user, 15, TRUE, src, BUSY_ICON_GENERIC))
```
Comment (psykzz) [inline code/game/objects/items/storage/belt.dm:346]:
```suggestion
			if(!do_after(user, 1.5 SECONDS, TRUE, src, BUSY_ICON_GENERIC))
```

**Pull request merged**